### PR TITLE
Add generic top_layer setting for blockage spacing.

### PR DESCRIPTION
**Pull request by colinschmidt**
Enables designs without floorplan information to still have blockage around macros.

### CODE REVIEW

**Comment by colinschmidt** [comment]
It's just the `top_layer` field of `placement_constraint`

**Comment by jwright6323** [comment]
Got it, and it is just specifying the top layer used by that block? Is there any reason not to reuse that for the blockage (or at least as a sane default if it is unspecified)?

**Comment by colinschmidt** [comment]
I think it is reasonable and perhaps even implied by the comment that it is what should be used.

**Comment by jwright6323** [comment]
Oh, I get it now. This is the global default and the others override.

File: src/hammer-vlsi/defaults.yml [hunk 1]
```diff
@@ -560,6 +560,12 @@ par:
   # type: Decimal
   blockage_spacing: 0.0
 
+  # Top metal layer that is pulled back around blocks and hardmacros
+  # Used by floorplanning.
+  # Overridden by individual placement constraints top_layer.
+  # type: Optional[String]
```
Comment (edwardcwang) [inline src/hammer-vlsi/defaults.yml]:
```suggestion
  # type: Optional[str]
```

**Comment by harrisonliew** [comment]
I'm a bit confused about this - you specify a layer to pull back globally across all blocks and hardmacros? Is the intention that doing when evoking the tool's auto-floorplanning utility, you don't have any `placement_constraints` and just want to see the impact of blocking off layers up to the specified one?

**Pull request merged**